### PR TITLE
Add request info to all RequestFailed exceptions

### PR DESCRIPTION
**Pull request by tsujigiri**
In the previous approach we missed a few of those.

### CODE REVIEW

File: spec/lib/soapy_cake/client_spec.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        .to receive(:start)
+        .and_yield(http)
+      allow(http)
+        .to receive(:request)
+        .and_return(response)
+    end
+
+    context 'receives non-200 HTTP status code' do
+      let(:response) do
+        instance_double(Net::HTTPInternalServerError, code: 500, body: '🔥')
```
Comment (RKushnir) [inline spec/lib/soapy_cake/client_spec.rb]: How about [webmock](response)?

File: lib/soapy_cake/client.rb [hunk 2]
```diff
@@ -29,8 +29,17 @@ def run(request)
...
-        xml_response? ? response.to_xml : response.to_enum
+        begin
+          response = Response.new(response_body(request), request.short_response?, time_converter)
+          xml_response? ? response.to_xml : response.to_enum
+        rescue RequestFailed => e
+          raise RequestFailed.new(
+            e.message,
+            request_path: request.path,
+            request_body: request.xml,
+            response_body: e.response_body || response.body
```
[the remark this answers is not in the document]
Comment (tsujigiri) [reply lib/soapy_cake/client.rb]: In that case the exception has it.

File: spec/lib/soapy_cake/client_spec.rb (hunk 1 above)
Comment (tsujigiri) [reply spec/lib/soapy_cake/client_spec.rb]: I'm just testing the logic between the two places a `RequestFailed` exception is raised and the information is attached. Integration tests already exist.

Comment (RKushnir) [reply spec/lib/soapy_cake/client_spec.rb]: But webmock does just the same what you do here, and wraps it in a better looking DSL.

File: lib/soapy_cake/client.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (RKushnir) [reply lib/soapy_cake/client.rb]: When does the exception not have it?

File: spec/lib/soapy_cake/client_spec.rb (hunk 1 above)
Comment (tsujigiri) [reply spec/lib/soapy_cake/client_spec.rb]: I'm not sure what you mean. Your link is broken.

File: lib/soapy_cake/client.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (tsujigiri) [reply lib/soapy_cake/client.rb]: I updated the reply above.

File: spec/lib/soapy_cake/client_spec.rb (hunk 1 above)
Comment (RKushnir) [reply spec/lib/soapy_cake/client_spec.rb]: Sorry, too much copy-pasting. This link https://rubygems.org/gems/webmock

File: lib/soapy_cake/client.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (RKushnir) [reply lib/soapy_cake/client.rb]: :+1:

File: spec/lib/soapy_cake/client_spec.rb (hunk 1 above)
Comment (tsujigiri) [reply spec/lib/soapy_cake/client_spec.rb]: Ah! Good idea! Updated!

**Pull request merged**